backend/api/Controllers/MissionController.cs

```diff
@@ -141,20 +216,114 @@ public async Task<ActionResult<byte[]>> GetMap([FromRoute] string assetCode, str
+            return StatusCode(StatusCodes.Status502BadGateway, $"{e.Message}");
+        }
+
+        List<MissionTask>? missionTasks;
+        switch (missionDefinition.Source.Type)
+        {
+            case MissionSourceType.Echo:
+                missionTasks = _echoService.GetMissionByPath(missionDefinition.Source.URL).Result.Tags
+                    .Select(
+                        t =>
+                        {
+                            var tagPosition = _stidService
+                                .GetTagPosition(t.TagId, missionDefinition.AssetCode)
+                                .Result;
+                            return new MissionTask(t, tagPosition);
+                        }
+                    )
+                    .ToList();
+                break;
+            case MissionSourceType.Custom:
+                missionTasks = _sourceService.GetMissionTasksFromURL(missionDefinition.Source.URL);
+                break;
+            default:
+                return BadRequest("Invalid mission source type provided");
+        }
+
+        if (missionTasks == null)
+            return NotFound("No mission tasks were found for the requested mission");
+
+        var scheduledMission = new MissionRun
+        {
+            Name = missionDefinition.Name,
+            Robot = robot,
+            MissionId = missionDefinition.Id,
+            Status = MissionStatus.Pending,
+            DesiredStartTime = rescheduledMissionQuery.DesiredStartTime,
+            Tasks = missionTasks,
+            AssetCode = missionDefinition.AssetCode,
+            Area = missionDefinition.Area,
+            Map = new MissionMap()
+        };
+
+        await _mapService.AssignMapToMission(scheduledMission);
+
+        if (scheduledMission.Tasks.Any())
+            scheduledMission.CalculateEstimatedDuration();
+
+        var newMission = await _missionRunService.Create(scheduledMission);
+
+        return CreatedAtAction(nameof(GetMissionRunById), new { id = newMission.Id }, newMission);
+    }
+
+    /// <summary>
+    /// Schedule a new echo mission
+    /// </summary>
+    /// <remarks>
+    /// <para> This query schedules a new echo mission and adds it to the database </para>
     /// </remarks>
     [HttpPost]
     [Authorize(Roles = Role.User)]
-    [ProducesResponseType(typeof(Mission), StatusCodes.Status201Created)]
+    [ProducesResponseType(typeof(MissionRun), StatusCodes.Status201Created)]
     [ProducesResponseType(StatusCodes.Status400BadRequest)]
     [ProducesResponseType(StatusCodes.Status401Unauthorized)]
     [ProducesResponseType(StatusCodes.Status403Forbidden)]
     [ProducesResponseType(StatusCodes.Status404NotFound)]
     [ProducesResponseType(StatusCodes.Status500InternalServerError)]
-    public async Task<ActionResult<Mission>> Create(
+    public async Task<ActionResult<MissionRun>> Create(
         [FromBody] ScheduledMissionQuery scheduledMissionQuery
```

Comment:
Consider deleting this endpoint as it would no longer be used.

Reply:
The simplest option would be to check if there exists a mission definition that uses the echo mission which the user is trying to schedule. In that case we can then reuse that mission definition. The problem arises when they want to use a different inspection frequency, or if they want to change it in other ways. An alternative which deals with this is to keep the MissionDefinitions separate from the echo missions, and expect users to first check MissionDefinitions for the echo mission they want before getting a new one from the echo API.